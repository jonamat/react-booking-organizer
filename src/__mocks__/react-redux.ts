import { Customer, Employee, Reservation, Service } from '../types';

export const __mockCustomers: Array<Customer> = [
    {
        id: 'customer-FAKE-ID1',
        name: 'Customer name',
        surname: 'Surname',
    },
    {
        id: 'customer-FAKE-ID2',
        name: 'Other name',
        surname: 'Other surname',
    },
];

export const __mockEmployees: Array<Employee> = [
    {
        id: 'employee-FAKE-ID1',
        name: 'Employee name 1',
    },
    {
        id: 'employee-FAKE-ID2',
        name: 'Employee name 2',
    },
];

export const __mockServices: Array<Service> = [
    {
        id: 'service-FAKE-ID1',
        name: 'fake service 1',
        averageDuration: 30,
    },
    {
        id: 'service-FAKE-ID2',
        name: 'fake service 2',
        averageDuration: 45,
    },
];

export const __mockReservations: Array<Reservation> = [
    {
        id: 'reservation-FAKE-ID1',
        date: new Date('1 Jan 2000 10:00 GMT+0100'),
        serviceId: __mockServices[0].id,
        customerId: __mockCustomers[0].id,
        employeeId: __mockEmployees[0].id,
        notes: 'something',
    },
    {
        id: 'reservation-FAKE-ID2',
        date: new Date('1 Jan 2040 10:00 GMT+0100'),
        serviceId: __mockServices[1].id,
        customerId: __mockCustomers[0].id,
        employeeId: __mockEmployees[1].id,
        notes: 'something2',
    },
];

export let useStore = () => ({
    getState: () => ({
        database: {
            customers: __mockCustomers,
            employees: __mockEmployees,
            services: __mockServices,
            reservations: __mockReservations,
        },
    }),
});

export const __setUseStore = (value) => (useStore = value);
export const useDispatch = () => undefined;
export const connect = () => (component) => component;
