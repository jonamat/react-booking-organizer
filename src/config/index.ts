import { WeekDays, HourMinute } from '../types';
let counter = 0;

/* -------------------------------------------------------------------------- */
/*                           Database configuration                           */
/* -------------------------------------------------------------------------- */

// IDs of the databases collections
export const RESERVATIONS_COLLECTION = 'reservations';
export const EMPLOYEES_COLLECTION = 'employees';
export const SERVICES_COLLECTION = 'services';
export const CUSTOMERS_COLLECTION = 'customers';

// Credentials for the sandbox database
export const SANDBOX_USERNAME = 'sandbox@demo.db';
export const SANDBOX_PASSWORD = 'sandbox';

/* -------------------------------------------------------------------------- */
/*                              App configuration                             */
/* -------------------------------------------------------------------------- */

// Key name of the employees order array in local storage
export const LS_EMPLOYEES_ORDER_KEY_NAME = 'employees_order';

// Minimum subdivision in minutes for the agenda. It should be a divisor of 60
export const MIN_STEP = 15;

// Hours of the longest service offered
export const MAX_SERVICE_DURATION_MINS = 3 * 60;

// Opening hour in HourMinute { hours(24), minutes }
export const OPENING_TIME: HourMinute = { hour: 8, minute: 0 };

// Closing hour in HourMinute { hours(24), minutes }
export const CLOSING_TIME: HourMinute = { hour: 20, minute: 0 };

/* -------------------------------------------------------------------------- */
/*                          Theming and localization                          */
/* -------------------------------------------------------------------------- */

// Progression of days according to the system
export const SYSTEM_ORDERED_DAYS: Array<WeekDays> = [
    'sunday',
    'monday',
    'tuesday',
    'wednesday',
    'thursday',
    'friday',
    'saturday',
];

// Progression of days according to the user local regulation (it_IT)
export const WEEK_DAYS: Array<WeekDays> = [
    'monday',
    'tuesday',
    'wednesday',
    'thursday',
    'friday',
    'saturday',
    'sunday',
];

// Reservation table sizes in px
export const CELLS_HEIGHT = 35;
export const CELLS_WIDTH = 200;
export const HEAD_COL_WIDTH = 70;

// Translations for material table
export const MATERIAL_TABLE_LOCALIZATION = {
    pagination: {
        labelRowsSelect: 'righe',
        labelRowsPerPage: 'Righe per pagina',
        firstAriaLabel: 'Prima pagina',
        firstTooltip: 'Prima pagina',
        previousAriaLabel: 'Pagina precedente',
        previousTooltip: 'Pagina precedente',
        nextAriaLabel: 'Pagina successiva',
        nextTooltip: 'Pagina successiva',
        lastAriaLabel: 'Ultima pagina',
        lastTooltip: 'Ultima pagina',
        labelDisplayedRows: '{from}-{to} di {count}',
    },
    toolbar: {
        searchTooltip: 'Cerca',
        searchPlaceholder: 'Cerca',
    },
    header: {
        actions: 'Azioni',
    },
    body: {
        emptyDataSourceMessage: 'Nessun elemento trovato',
        filterRow: {
            filterTooltip: 'Filtra',
        },
    },
};

// App date format
export const TIME_DATE_FORMAT = {
    weekday: 'long',
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: 'numeric',
    minute: 'numeric',
};

/* -------------------------------------------------------------------------- */
/*                    Constants derived from previous vars                    */
/* -------------------------------------------------------------------------- */

// Steps which has 1 hour
export const SEGMENTS_IN_1HOUR = 60 / MIN_STEP;
counter = 0;

// Array of progressive minutes steps within a hour
export const PROGRESSIVE_MINS_WITHIN_HOUR = new Array<number>(SEGMENTS_IN_1HOUR)
    .fill(0)
    .map(() => MIN_STEP * counter++);
counter = 0;

// Array of progressive minutes steps within MAX_SERVICE_DURATION_MINS without the first element (zero)
export const PROGRESSIVE_MINS_WITHIN_MAX_SERVICE = new Array<number>(
    Math.floor(MAX_SERVICE_DURATION_MINS / MIN_STEP) + 1,
)
    .fill(0)
    .map(() => MIN_STEP * counter++)
    .slice(1);
counter = 0;

// Array of progressive HourMinute steps within a business day
export const PROGRESSIVE_HOURS_WITHIN_BUSINESS_DAY: Array<HourMinute> = new Array<string>(
    Math.floor(
        (Math.abs(CLOSING_TIME.hour - OPENING_TIME.hour) * 60 +
            Math.abs(CLOSING_TIME.minute - OPENING_TIME.minute) +
            MIN_STEP) /
            MIN_STEP,
    ),
)
    .fill('')
    .map(() => {
        const totalMinutes = MIN_STEP * counter++;
        const hour = Math.floor(totalMinutes / 60) + OPENING_TIME.hour;
        const minute = (totalMinutes % 60) + OPENING_TIME.minute;
        return { hour, minute };
    });
