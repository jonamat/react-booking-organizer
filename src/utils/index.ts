// Utils
export { default as isEmployeeBusy } from './isEmployeeBusy';
export { default as isEmployeeOnDuty } from './isEmployeeOnDuty';
export { default as generateClosestValidDate } from './generateClosestValidDate';
export { default as getReservationsByDate } from './getReservationsByDate';
export { default as getReservationStatus } from './getReservationStatus';

export { default as formatMinutes } from './formatMinutes';
export { default as hourMinuteToReadableString } from './hourMinuteToReadableString';
export { default as translateWeekDays } from './translateWeekDays';

export { default as databaseSync } from './databaseSync';
export { default as connectionListener } from './connectionListener';

// Auth system
export { default as login } from './login';
export { default as logout } from './logout';

// Delete database elements
export { default as deleteCustomer } from './deleteCustomer';
export { default as deleteReservation } from './deleteReservation';
export { default as deleteEmployee } from './deleteEmployee';
export { default as deleteService } from './deleteService';

// Add database elements
export { default as addCustomer } from './addCustomer';
export { default as addEmployee } from './addEmployee';
export { default as addReservation } from './addReservation';
export { default as addService } from './addService';

// Edit database elements
export { default as editCustomer } from './editCustomer';
export { default as editEmployee } from './editEmployee';
export { default as editReservation } from './editReservation';
export { default as editService } from './editService';
