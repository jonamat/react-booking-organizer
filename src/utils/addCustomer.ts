import firebase from 'firebase/app';

import { NewCustomer, isNewCustomer, Customer } from '../types';
import { CUSTOMERS_COLLECTION } from '../config';

const REQUIRED_FIELDS = ['name'];

/**
 * Validate user inputs and send them to database
 * @param formData Raw data from form
 * @param birthday Birthday Date instance
 */
const addCustomer = async (formData: FormData, birthday: Date | null): Promise<Customer> => {
    if (REQUIRED_FIELDS.some((fieldName) => !formData.get(fieldName))) throw new Error('Compila i campi obbligatori');

    const newCustomer: any = {
        name: formData.get('name') || undefined,
        surname: formData.get('surname') || undefined,
        additionalIdentifier: formData.get('additionalIdentifier') || undefined,
        phone: formData.get('phone') || undefined,
        birthday: birthday || undefined,
        notes: formData.get('notes') || undefined,
    };

    if (!isNewCustomer(newCustomer)) throw new Error('Errore interno. Contatta il supporto tecnico');

    // Delete empty values
    (Object.keys(newCustomer) as Array<keyof NewCustomer>).forEach((key) => {
        if (newCustomer[key] === undefined) delete newCustomer[key];
    });

    try {
        return {
            id: (await firebase.firestore().collection(CUSTOMERS_COLLECTION).add(newCustomer)).id,
            ...newCustomer,
        };
    } catch (error) {
        console.error(error);
        throw new Error('Connessione al database fallita');
    }
};

export default addCustomer;
