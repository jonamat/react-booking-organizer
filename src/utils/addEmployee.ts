import firebase from 'firebase/app';

import { NewEmployee, Employee, isNewEmployee, WeekWorkingShifts, Holiday } from '../types';
import { EMPLOYEES_COLLECTION } from '../config';

const REQUIRED_FIELDS = ['name'];

/**
 * Validate user inputs and send them to database
 * @param formData Raw data from form
 * @param weekWorkingShifts Valid WeekWorkingShift object
 * @param holidays Valid Holiday array
 */
const addEmployee = async (
    formData: FormData,
    weekWorkingShifts?: WeekWorkingShifts | null,
    holidays?: Array<Holiday> | null,
): Promise<Employee> => {
    if (REQUIRED_FIELDS.some((fieldName) => !formData.get(fieldName))) throw new Error('Compila i campi obbligatori');

    const newEmployee: any = {
        name: formData.get('name') || undefined,
        weekWorkingShifts,
        holidays,
    };

    if (!isNewEmployee(newEmployee)) throw new Error('Errore interno. Contatta il supporto tecnico');

    (Object.keys(newEmployee) as Array<keyof NewEmployee>).forEach((key) => {
        if (newEmployee[key] === undefined) delete newEmployee[key];
    });

    try {
        return {
            id: (await firebase.firestore().collection(EMPLOYEES_COLLECTION).add(newEmployee)).id,
            ...newEmployee,
        };
    } catch (error) {
        console.error(error);
        throw new Error('Connessione al database fallita');
    }
};

export default addEmployee;
