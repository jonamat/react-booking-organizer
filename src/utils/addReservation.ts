import { Employee, Service, Customer, isNewReservation, NewReservation, Reservation } from '../types';
import firebase from 'firebase/app';
import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';

import { PROGRESSIVE_MINS_WITHIN_HOUR, MIN_STEP, TIME_DATE_FORMAT, RESERVATIONS_COLLECTION } from '../config';
import { isEmployeeBusy, isEmployeeOnDuty } from '.';

const REQUIRED_FIELDS = ['serviceId'];

interface Database {
    employees: Array<Employee>;
    services: Array<Service>;
    reservations: Array<Reservation>;
}

/**
 * Validate user inputs and send them to database
 * @param formData Raw data from form
 * @param customer Customer requesting the reservation
 * @param selectedDate Date of the reservation
 * @param validationData Redux data from store.database for validation
 * @param forceAssignment Force assignment even if employee in not on duty
 */
const addReservation = async (
    formData: FormData,
    customer: Customer | null,
    selectedDate: MaterialUiPickersDate,
    { employees, services, reservations }: Database,
    forceAssignment?: boolean,
): Promise<Reservation> => {
    if (REQUIRED_FIELDS.some((fieldName) => !formData.get(fieldName)) || !customer)
        throw new Error('Compila i campi obbligatori');

    if (!selectedDate || (typeof selectedDate === 'number' && isNaN(selectedDate)))
        throw new Error('Seleziona una data valida');

    if (!PROGRESSIVE_MINS_WITHIN_HOUR.includes(selectedDate.getMinutes()))
        throw new Error(`L'orario indicano non rispetta lo scaglionamento orario dell'agenda.
        Utilizza intervalli di ${MIN_STEP} minuti.`);

    const newReservation: any = {
        date: selectedDate,
        customerId: customer.id,
        serviceId: formData.get('serviceId') || undefined,
        employeeId: formData.get('employeeId') || undefined,
        notes: formData.get('notes') || undefined,
    };

    if (!isNewReservation(newReservation)) throw new Error('Errore interno. Contatta il supporto tecnico');

    const humanReadableDate = newReservation.date.toLocaleDateString('it-IT', TIME_DATE_FORMAT);

    const employee = employees.find((employee) => employee.id === newReservation.employeeId);
    if (!employee) throw new Error('Errore interno - Dipendente non trovato');

    if (isEmployeeBusy(newReservation, services, reservations))
        throw new Error(`${employee?.name} è già impegnato/a con un altro appuntamento ${humanReadableDate}.`);

    if (!forceAssignment && !isEmployeeOnDuty(newReservation.date, employee)) throw new Error('EMPLOYEE_NOT_ON_DUTY');

    (Object.keys(newReservation) as Array<keyof NewReservation>).forEach((key) => {
        if (newReservation[key] === undefined) delete newReservation[key];
    });

    try {
        return {
            id: (await firebase.firestore().collection(RESERVATIONS_COLLECTION).add(newReservation)).id,
            ...newReservation,
        };
    } catch (error) {
        console.error(error);
        throw new Error('Connessione al database fallita');
    }
};

export default addReservation;
