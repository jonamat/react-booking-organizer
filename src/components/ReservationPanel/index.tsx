import React, { FC, useEffect, useState } from 'react';
import { Typography, Paper } from '@material-ui/core';
import { useStore } from 'react-redux';
import { withRouter, RouteComponentProps } from 'react-router';

import { Customer, Reservation, RootState } from '../../types';
import { MIN_STEP, TIME_DATE_FORMAT } from '../../config';
import { getReservationStatus } from '../../utils';
import useStyle from './style';

interface ReservationsTableProps extends RouteComponentProps {
    reservation: Reservation | undefined;
    showDetails?: boolean;
}

export const composeName = (customer: Customer): string =>
    customer.name +
    (customer.surname ? ' ' + customer.surname : '') +
    (customer.additionalIdentifier ? ' ' + customer.additionalIdentifier : '');

const ReservationPanel: FC<ReservationsTableProps> = ({ reservation, showDetails, history }) => {
    const {
        database: { services, customers },
    } = useStore<RootState>().getState();

    if (!reservation || !services || !customers) throw new Error('Internal error - Resources loading logic broken');

    // Re-render every minute to check the reservation status and adjust the bg color
    const [, update] = useState({});
    useEffect(() => {
        const updateForStatus = setInterval(() => update({}), 1000 * 60);
        return () => clearInterval(updateForStatus);
    }, []);

    const service = services.find((service) => service.id === reservation.serviceId);
    const customer = customers.find((customer) => customer.id === reservation.customerId);

    const handleReservationDetails = () => history.push('/reservation-details/' + reservation.id);

    // Style variants
    const reservationStatus = service ? getReservationStatus(reservation.date, service.averageDuration) : 'ERROR';
    const occupiedCellsSpace = showDetails ? 4 : service ? service.averageDuration / MIN_STEP : 1;

    const classes = useStyle({ occupiedCellsSpace, showDetails, reservationStatus });

    return (
        <Paper className={classes.container} onClick={handleReservationDetails}>
            <div className={classes.spacing}>
                {showDetails && (
                    <Typography variant="subtitle2" className={classes.date} data-testid="date">
                        {reservation.date.toLocaleDateString('it-IT', TIME_DATE_FORMAT)}
                    </Typography>
                )}
                {customer && (
                    <Typography variant="body2" className={classes.name} data-testid="name">
                        {composeName(customer)}
                    </Typography>
                )}
                {(!service || !customer) && (
                    <Typography variant="body2" className={classes.ongoing} data-testid="missing-info">
                        Informazioni mancanti
                    </Typography>
                )}
                {reservationStatus === 'ONGOING' && (
                    <Typography variant="body2" className={classes.ongoing} data-testid="ongoing-now">
                        In corso
                    </Typography>
                )}
                {service && (
                    <Typography variant="caption" className={classes.service} data-testid="service">
                        {service.name}
                    </Typography>
                )}
                {reservation.notes && (
                    <Typography variant="body2" data-testid="notes">
                        {reservation.notes}
                    </Typography>
                )}
            </div>
        </Paper>
    );
};

export default withRouter(ReservationPanel);
