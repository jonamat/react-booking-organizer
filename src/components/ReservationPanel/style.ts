import { makeStyles, Theme } from '@material-ui/core';
import { CELLS_HEIGHT } from '../../config';
import { getReservationStatus } from '../../utils';

type BackgroundColorCodes = ReturnType<typeof getReservationStatus> | 'ERROR';

const createBackgroundColor = (status: BackgroundColorCodes): string => {
    switch (status) {
        case 'DONE':
            return '#d8d8d8';
        case 'ONGOING':
            return '#f5eec4';
        case 'UNINITIATED':
            return '#c4c8f5';
        case 'ERROR':
            return '#ffa0a0';
    }
};

export const useStyles = makeStyles<
    Theme,
    { showDetails?: boolean; occupiedCellsSpace: number; reservationStatus: BackgroundColorCodes }
>((theme) => ({
    container: {
        position: 'relative',
        top: 3,
        zIndex: 1,
        overflow: 'auto',
        margin: 'auto',
        height: ({ occupiedCellsSpace }): number =>
            CELLS_HEIGHT * occupiedCellsSpace - /* top-bottom space */ 6 - /* box shadow */ 1,
        width: ({ showDetails }): string => (showDetails ? 'auto' : '95%'),
        transition: theme.transitions.create('background-color'),
        cursor: 'pointer',
        backgroundColor: ({ reservationStatus }): string => createBackgroundColor(reservationStatus),
        '&:hover': {
            backgroundColor: '#d1d5ee',
        },
    },
    spacing: {
        padding: theme.spacing(1),
    },
    date: {
        color: '#f50057',
    },
    name: {
        fontWeight: theme.typography.fontWeightMedium,
    },
    ongoing: {
        fontWeight: theme.typography.fontWeightMedium,
        color: '#f44336',
        textTransform: 'uppercase',
    },
    service: {
        color: '#0020a8',
    },
}));

export default useStyles;
