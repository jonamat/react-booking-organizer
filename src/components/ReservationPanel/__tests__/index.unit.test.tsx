/**
 * Unit test suite
 *
 * Target: ReservationPanel component
 * Tests:
 * - Data position correctness
 * - Redirection
 * - Snapshot matching
 * - Functionality
 */

// Test resources
import React, { ComponentProps } from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import { StylesProvider, createGenerateClassName } from '@material-ui/core';

// Test configuration
import { __mockReservations, __mockCustomers, __mockServices } from '../../../__mocks__/react-redux';

// Mocked modules
import { getReservationStatus } from '../../../utils';

// Component to test
import ReservationPanel, { composeName } from '../';
import { TIME_DATE_FORMAT } from '../../../config';

/* -------------------------------------------------------------------------- */
/*                                    Mocks                                   */
/* -------------------------------------------------------------------------- */

jest.mock('react-router', () => ({
    withRouter: (component) => component,
}));

jest.mock('../../../utils', () => ({
    getReservationStatus: jest.fn(() => 'DONE'),
}));
const history = {
    push: jest.fn(),
};

/* -------------------------------------------------------------------------- */
/*                              Bridge operations                             */
/* -------------------------------------------------------------------------- */

beforeEach(() => {
    jest.clearAllMocks();
});
/* -------------------------------------------------------------------------- */
/*                               Render wrapper                               */
/* -------------------------------------------------------------------------- */

/**
 * Render the component with the custom configuration
 * @param props Component props
 */
const renderComponent = (props?: Partial<ComponentProps<typeof ReservationPanel>>) => {
    return render(
        <ReservationPanel
            // @ts-ignore
            history={history}
            reservation={__mockReservations[0]}
            showDetails={false}
            {...props}
        />,
    );
};

/* -------------------------------------------------------------------------- */
/*                                    Tests                                   */
/* -------------------------------------------------------------------------- */

describe('ReservationPanel', () => {
    describe('Status progress', () => {
        it('calculated by calling getReservationStatus', () => {
            renderComponent();
            expect(getReservationStatus as jest.Mock).toBeCalledTimes(1);
        });

        it('refreshes every minute', () => {
            jest.useFakeTimers();
            renderComponent();

            // First call on mount
            expect(getReservationStatus as jest.Mock).toBeCalledTimes(1);

            act(() => {
                jest.advanceTimersByTime(1000 * 60);
            });
            expect(getReservationStatus as jest.Mock).toBeCalledTimes(2);
            act(() => {
                jest.advanceTimersByTime(1000 * 60);
            });
            expect(getReservationStatus as jest.Mock).toBeCalledTimes(3);
        });
    });

    describe('Data accuracy', () => {
        it('shows correct data in correct position', () => {
            const { queryByTestId } = renderComponent({ showDetails: true });

            expect(queryByTestId('date')).toHaveTextContent(
                __mockReservations[0].date.toLocaleDateString('it-IT', TIME_DATE_FORMAT),
            );
            expect(queryByTestId('name')).toHaveTextContent(composeName(__mockCustomers[0]));
            expect(queryByTestId('missing-info')).toBeFalsy();
            expect(queryByTestId('ongoing-now')).toBeFalsy();
            expect(queryByTestId('service')).toHaveTextContent(__mockServices[0].name);
            expect(queryByTestId('notes')).toHaveTextContent(__mockReservations[0].notes as string);
        });
    });

    describe('User experience', () => {
        it('redirects users to reservation details on panel click', () => {
            const { getByTestId } = renderComponent();

            fireEvent.click(getByTestId('name'));

            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith('/reservation-details/' + __mockReservations[0].id);
        });
    });

    it('matches snapshot', () => {
        const history = createMemoryHistory();
        const snapshot = renderer
            .create(
                <StylesProvider generateClassName={createGenerateClassName({ disableGlobal: true })}>
                    {/* @ts-ignore */}
                    <ReservationPanel history={history} reservation={__mockReservations[0]} showDetails={true} />
                </StylesProvider>,
            )
            .toJSON();
        expect(snapshot).toMatchSnapshot();
    });
});
