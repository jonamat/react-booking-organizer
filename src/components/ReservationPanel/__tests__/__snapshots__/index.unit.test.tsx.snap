// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationPanel matches snapshot 1`] = `
<div
  className="jss8 jss1 jss7 jss12 jss9"
  onClick={[Function]}
>
  <div
    className="jss2"
  >
    <h6
      className="jss36 jss3 jss48"
      data-testid="date"
    >
      sabato 1 gennaio 2000, 10:00
    </h6>
    <p
      className="jss36 jss4 jss37"
      data-testid="name"
    >
      Customer name Surname
    </p>
    <span
      className="jss36 jss6 jss39"
      data-testid="service"
    >
      fake service 1
    </span>
    <p
      className="jss36 jss37"
      data-testid="notes"
    >
      something
    </p>
  </div>
</div>
`;
